src/codeu/chat/server/Server.java

```diff
@@ -112,6 +128,26 @@ public void onMessage(InputStream in, OutputStream out) throws IOException {
 
         final String name = Serializers.STRING.read(in);
         final User user = controller.newUser(name);
+        
+        //if a user can be created
+        //then we will add the command to the queue
+        if(user != null){
+        	String userAddCommand = "U-ADD ";
+            userAddCommand += user.id.id() + " ";
+            userAddCommand += user.name + " ";
+            userAddCommand += user.creation.inMs();
+

```

Comment:
Style-wise: I would not reassign with +=, but just have + with line separations.